arbnode/dataposter/data_poster.go

```diff
@@ -175,22 +175,32 @@ func NewDataPoster(ctx context.Context, opts *DataPosterOpts) (*DataPoster, erro
 }
 
 func rpcClient(ctx context.Context, opts *ExternalSignerCfg) (*rpc.Client, error) {
-	rootCrt, err := os.ReadFile(opts.RootCA)
+	clientCert, err := tls.LoadX509KeyPair(opts.ClientCert, opts.ClientPrivateKey)
```

Comment:
I don't think running in such a configuration would be generally advisable, but could we make the client certificate optional just in case?

Reply:
Also, can we default to reading the private key from the cert file if the private key option isn't separately specified? It's just a slight quality of life improvement, as often times the cert and key are in the same file.

Reply:
Changed client certificate to be optional.

As for private key being in the same file as cert file, I think this is generally discouraged since private key and certificate files should generally have different access controls. Although if we have justification why we need it anyway, I'll add it.